flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/TimeFunctionsITCase.java

```diff
@@ -425,4 +430,304 @@ private Stream<TestSetSpec> temporalOverlapsTestCases() {
                                 temporalOverlaps($("f1"), $("f0"), $("f1"), $("f0")),
                                 "TEMPORAL_OVERLAPS requires 2nd argument 'leftTemporal' to be DATETIME or INTERVAL type, but is INT"));
     }
+
+    private Stream<TestSetSpec> ceilTestCases() {
+        return Stream.of(
+                TestSetSpec.forFunction(BuiltInFunctionDefinitions.FLOOR)
+                        .onFieldsWithData(
+                                LocalTime.of(11, 22, 33),
```

Comment:
Maybe we can have a time which also has nanos, to test the truncation to millis, seconds, etc.

Reply:
True, please add the following comment, here and for the ceiling:
```
// https://issues.apache.org/jira/browse/FLINK-17224 Fractional seconds are lost
```